src/datadog/grok/parse_grok.rs

```diff
@@ -51,19 +51,15 @@ fn apply_grok_rule(source: &str, grok_rule: &GrokRule) -> Result<Value, Error> {
                     filters,
                 }) = grok_rule.fields.get(name)
                 {
-                    filters.iter().for_each(|filter| {
-                    if let Some(ref mut v) = value {
-                        value = match apply_filter(v, filter) {
-                            Ok(Value::Null) => None,
-                            Ok(v) if v.is_object() => Some(parse_keys_as_path(v)),
-                            Ok(v) => Some(v),
-                            Err(error) => {
-                                warn!(message = "Error applying filter", field = %field, filter = %filter, %error);
-                                None
-                            }
-                        };
+                    for filter in filters {
+                        if let Some(ref mut v) = value {
+                            value = match apply_filter(v, filter)? {
```

Comment:
In the DD Logs parsing, if a filter fails, the rest of the parsed values are returned e.g. 
- input: `word not_an_integer another_word`
- rule `%{word:field1} %{data:field2:integer} %{word:field3}`
- result `{"field1": "word", "field3": "another_word"}`

I'm thinking that might be why we were returning `None` in the error case. If we return early in err I think we lose that, based on this unit test I used
```
#[test]
    fn continue_if_filter_fails() {
        test_full_grok(vec![
            // not a number
            (
                "%{word:field1} %{data:field2:integer} %{word:field3}",
                "word not_an_integer another_word",
                Ok(Value::from(btreemap! {
                    "field1" => Value::Bytes("word".into()),
                    "field3" => Value::Bytes("another_word".into()),
                })),
            ),
        ]);
    }
```

Reply:
btw for the Logs behavior I checked using a Grok Parser [here](https://dd.datad0g.com/logs/pipelines?search=team%3Aobservability_pipelines) in staging. I submitted some actual logs too but the sample testing UI seems reliable enough

Reply:
👋  thanks for calling this out. I've adjusted the change to differentiate between fatal and non fatal errors to let the caller decide.